http/oidc/src/main/java/org/wildfly/security/http/oidc/LogoutHandler.java

```diff
@@ -0,0 +1,313 @@
+                .build();
+
+        JwtClaims claims = null;
+        Exception cause = null;
+        try {
+            // check keycloak 'typ'
+            claims = tokenValidator.verify(logoutToken);
+        } catch (Exception expKeyclockClaims) {
+            cause = expKeyclockClaims;
+            if (expKeyclockClaims.getCause().getMessage().contains("ELY23054: Unexpected value for typ claim")) {
+                log.warn("OpenID Provider claims typ " + providerLogoutTokenType
+                        + " was not valid.  Trying typ "+ LOGOUT_JWT_TOKEN_TYPE);
+
+                // check other OP's 'typ'
+                tokenValidator = tokenBuilder.setTokenType(LOGOUT_JWT_TOKEN_TYPE)
+                                .build();
+                try {
+                    claims = tokenValidator.verify(logoutToken);
+                } catch (Exception expOtherProviderCliams) {
+                    cause = expOtherProviderCliams;
+                }
+            }
+            if (claims == null) {
+                log.debugf("Unexpected error when verifying logout token", cause);
+                facade.getResponse().setStatus(HttpStatus.SC_BAD_REQUEST);
+                facade.authenticationFailed();
+                return;
+            }
+        }
+
+        if (!isLogoutSessionRequired(facade)) {
+            log.warn(log.sidCanNotBeInferFromLogoutToken());
+            facade.getResponse().setStatus(HttpStatus.SC_BAD_REQUEST);
+            facade.authenticationFailed();
+            return;
+        }
+
+        String sessionId = claims.getClaimValueAsString(SID);
+
+        if (sessionId == null) {
+            facade.getResponse().setStatus(HttpStatus.SC_BAD_REQUEST);
+            facade.authenticationFailed();
+            return;
+        }
+
+        log.debugf("Marking session for invalidation during back-channel logout");
+        sessionsMarkedForInvalidation.put(getSessionKey(facade, sessionId), facade.getOidcClientConfiguration());
+    }
+
+    private String getSessionKey(OidcHttpFacade facade, String sessionId) {
+        return facade.getOidcClientConfiguration().getClientId() + CLIENT_ID_SID_SEPARATOR + sessionId;
+    }
+
+    private void handleFrontChannelLogoutRequest(OidcHttpFacade facade) {
+        if (isLogoutSessionRequired(facade)) {
+            Request request = facade.getRequest();
+            String sessionId = request.getQueryParamValue(SID);
+
+            if (sessionId == null) {
+                facade.getResponse().setStatus(HttpStatus.SC_BAD_REQUEST);
+                facade.authenticationFailed();
+                return;
+            }
+
+            RefreshableOidcSecurityContext context = getSecurityContext(facade);
+            IDToken idToken = context.getIDToken();
+            String issuer = request.getQueryParamValue(ISS);
+
+            if (idToken == null || !sessionId.equals(idToken.getNonce()) || !idToken.getIssuer().equals(issuer)) {
```

Comment:
this is another instance of comparing `sessionId` (from `sid` query parameter) to `idToken.getNonce()` . Not sure  why since these are different concepts it seems to me. I wonder if Keycloak configures these values to match, but other providers might not match them. we will need to investigate this